src/structlog/processors.py

```diff
@@ -531,3 +543,184 @@ def __call__(
+    """
+
+    #: The full path to the python source file of the callsite.
+    PATHNAME = "pathname"
+    #: The basename part of the full path to the python source file of the
+    #: callsite.
+    FILENAME = "filename"
+    #: The python module the callsite was in. This mimicks the module attribute
+    #: of `logging.LogRecord` objects and will be the basename, without
+    #: extension, of the full path to the python source file of the callsite.
+    MODULE = "module"
+    #: The name of the function that the callsite was in.
+    FUNC_NAME = "func_name"
+    #: The line number of the callsite.
+    LINENO = "lineno"
+    #: The ID of the thread the callsite was executed in.
+    THREAD = "thread"
+    #: The name of the thread the callsite was executed in.
+    THREAD_NAME = "thread_name"
+    #: The ID of the process the callsite was executed in.
+    PROCESS = "process"
+    #: The name of the process the callsite was executed in.
+    PROCESS_NAME = "process_name"
+
+
+class CallsiteParameterAdder:
+    """
+    Adds parameters of the callsite that an event dictionary originated from to
+    the event dictionary. This processor can be used to enrich events
+    dictionaries with information such as the function name, line number and
+    filename that an event dictionary originated from.
+
+    .. warning::
+        This processor cannot detect the correct callsite for invocation of
+        async functions.
+
+    If the event dictionary has an embedded `logging.LogRecord` object and did
+    not originate from `structlog` then the callsite information will be
+    determined from the `logging.LogRecord` object. For event dictionaries
+    without an embedded `logging.LogRecord` object the callsite will be
+    determined from the stack trace, ignoring all intra-structlog calls, calls
+    from the `logging` module, and stack frames from modules with names that
+    start with values in ``additional_ignores``, if it is specified.
+
+    The keys used for callsite parameters in the event dictionary are the
+    string values of `CallsiteParameter` enum members.
+
+    :param parameters:
+        A collection of `CallsiteParameter` values that should be added to the
+        event dictionary.
+
+    :param additional_ignores:
+        Additional names with which the a stack frame's module name must not
+        start for it to be considered when determening the callsite.
+
+    .. note::
+        When used with `structlog.stdlib.ProcessorFormatter` the most efficient
+        configuration is to either use this processor in ``foreign_pre_chain``
+        of `structlog.stdlib.ProcessorFormatter` and in ``processors`` of
+        `structlog.configure`, or to use it in ``processor`` of
+        `structlog.stdlib.ProcessorFormatter` without using it in
+        ``processors`` of `structlog.configure` and ``foreign_pre_chain`` of
+        `structlog.stdlib.ProcessorFormatter`.
+
+    .. versionadded:: 21.5.0
+    """
+
+    _handlers: Dict[
```

Comment:
should this be a ClassVar too?

Reply:
yes, same for `_all_parameters`, will change both.

Reply:
Changed `_handlers` and `_all_parameters` to `ClassVar`.